Add organization-specific detections

shinenelson opened the pull request:
GitHub refreshed organization profiles and put up more organization-wide tabs on the organization profile with dedicated routes to each of the tabs ( unlike user profiles ).

This change-set also renames `isOrganizationDiscussion` to `isOrganizationTeamDiscussion` since 'discussions' are now a core public feature and not only restricted to organization teams. And moreover, the organization team discussions are kind of disjoint from the public feature.

There are 2 other detections that would depend on `isOrganizationRepo`. I was not sure whether to include them.
- `isOrganizationPackage`
- `isOrganizationDiscussion`

Both of these detections eventually redirect into a organization repository.

Organization packages do have a URL in the `/orgs` namespace, but since they are eventually redirected to an organization repository, I am not sure whether it is worth 'detecting'. Maybe, it might be a dedicated page in the future? :thinking:

fregante commented:
Do you plan on using these on GitHub? I don't generally add detections "just in case" because they might break before anyone even uses them

shinenelson commented:
The intention was to use it for https://github.com/sindresorhus/refined-github/pull/4743 to have [specific includes](https://github.com/sindresorhus/refined-github/pull/4743#discussion_r698044916) on the organization pages.

What I wanted was for repositories and people, but I admit I added the detection for packages, projects and discussion _just because_ even though I did not have any immediate use for them. I added them since I was testing with the organization pages anyway.

fregante commented on index.ts:
```diff
@@ -163,14 +163,48 @@ collect.set('isNotifications', [
 	'https://github.com/orgs/refined-github/teams/core-team',
 ]);
+/** @deprecated use isOrganizationTeamDiscussion instead */
+export const isOrganizationDiscussion = isOrganizationTeamDiscussion;
```
Tests are failing because this still needs example URLs. I don't remember how to copy them, but I think this happened before so you can look in this file

shinenelson replied:
Yes, I copied from `50bd30f` which did not take URLs arguments ( something I realized only after I looked more closely after seeing the test fail ).

fregante commented on index.ts:
```diff
@@ -163,14 +163,48 @@ collect.set('isNotifications', [
+
+export const isOrganizationDiscussionsList = (url: URL | HTMLAnchorElement | Location = location): boolean => /^orgs\/[^/]+\/discussions$/.test(getCleanPathname(url));
+collect.set('isOrganizationDiscussionsList', [
+	'https://github.com/orgs/github/discussions',
```
More than one URL is needed. Are these pages paginated? Add page 2 etc. If not, look for other types of variations. If they're not used they can be dropped for now since they're simple to be implement _if and when_ they're needed.

shinenelson replied:
It is difficult to find organizations that use discussions globally. And the ones that I did find did not have pagination.

I will drop this for now.

fregante commented on index.ts:
```diff
@@ -163,14 +163,48 @@ collect.set('isNotifications', [
+	'https://github.com/orgs/github/packages',
+]);
+
+export const isOrganizationPeopleList = (url: URL | HTMLAnchorElement | Location = location): boolean => /^orgs\/[^/]+\/people$/.test(getCleanPathname(url));
```
```suggestion
export const isOrganizationMembers = (url: URL | HTMLAnchorElement | Location = location): boolean => /^orgs\/[^/]+\/people$/.test(getCleanPathname(url));
```

There's no "single member" page so it doesn't need "list"

fregante commented on index.ts:
```diff
@@ -163,14 +163,48 @@ collect.set('isNotifications', [
 
-export const isOrganizationDiscussion = (url: URL | HTMLAnchorElement | Location = location): boolean => /^orgs\/[^/]+\/teams\/[^/]+($|\/discussions)/.test(getCleanPathname(url));
-collect.set('isOrganizationDiscussion', [
+export const isOrganizationPackagesList = (url: URL | HTMLAnchorElement | Location = location): boolean => /^orgs\/[^/]+\/packages$/.test(getCleanPathname(url));
```
**singular**List like `isConversationList`

```suggestion
export const isOrganizationPackageList = (url: URL | HTMLAnchorElement | Location = location): boolean => /^orgs\/[^/]+\/packages$/.test(getCleanPathname(url));
```

This applies to every other detection

fregante reviewed (commented):
I published the current code as prerelease v5.4.0-0 so you can use it in the Refined GitHub repo. I'll leave this open a bit longer

shinenelson commented:
> I published the current code as prerelease v5.4.0-0 so you can use it in the Refined GitHub repo.

I do not see the tag or release in the repository. But either way, I do not think it would have code from this pull request. So there would not be anything useful that I could use to test.

Or were you implying something else when you said _use it in the Refined GitHub repo_?

fregante commented:
It's on npm only, it's everything you see in this PR, you have to install the version specifically, it's a prerelease

shinenelson commented:
> It's on npm only, it's everything you see in this PR, you have to install the version specifically, it's a prerelease

Ah, okay, that's great :+1: Thanks.

fregante commented:
Drafted until https://github.com/sindresorhus/refined-github/pull/4743 is merged and CI is fixed

fregante commented:
https://github.com/refined-github/refined-github/pull/4743 was merged, can you finish this PR?
